### ISSUE TITLE
Multiple angular2-multiselect-dropdowns throw error

### ISSUE THREAD

**Issue by Mridul16** [CuppaLabs/angular2-multiselect-dropdown#77]
I have been trying to add 2 angular2-multiselect-dropdown on the same page but I am getting this error-
Only arrays and iterables are allowed at DefaultIterableDiffer.webpackJsonp.../../../core/@angular/core.es5.js.DefaultIterableDiffer.diff 

Please help

**Comment by CuppaLabs** [comment]
You data structure might not be appropriate. Please refer to demo page https://cuppalabs.github.io/angular2-multiselect-dropdown/. The demo page has multiple dropdown in one page. Working as expected. Hence closing the issue. Also refer to code at https://github.com/CuppaLabs/angular2-multiselect-dropdown/blob/master/src/app/app.component.ts

**Comment by Mridul16** [comment]
This is the code m using

<div class="select_category pull-left col-sm-6 col-xs-12 paddingZero">
                    <div class="contianer_icon">
                        <img src="{{selectedCategory.image}}" />
                    </div>
                    <div class="select_box">
                        <angular2-multiselect [data]="dropdownList" [(ngModel)]="selectedService" [settings]="dropdownSettings" (onSelect)="onItemSelect($event)" (onDeSelect)="OnItemDeSelect($event)"></angular2-multiselect>
                    </div>
                </div>
                <div class="select_category pull-left col-sm-6 col-xs-12 paddingZero">
                    <div class="contianer_icon">
                        <i class="fa fa-map-marker" aria-hidden="true"></i>
                    </div>
                    <div class="select_box">
                        <angular2-multiselect [data]="cityDropdownList" [(ngModel)]="selectedLocation" [settings]="cityDropdownSettings" (onSelect)="onItem1Select($event)" (onDeSelect)="OnItem1DeSelect($event)"></angular2-multiselect>
                    </div>
                </div>

**Comment by Mridul16** [comment]
'<div class="select_category pull-left col-sm-6 col-xs-12 paddingZero">
                    <div class="contianer_icon">
                        <img src="{{selectedCategory.image}}" />
                    </div>
                    <div class="select_box">
                        <angular2-multiselect [data]="dropdownList" [(ngModel)]="selectedService" [settings]="dropdownSettings" (onSelect)="onItemSelect($event)" (onDeSelect)="OnItemDeSelect($event)"></angular2-multiselect>
                    </div>
                </div>
                <div class="select_category pull-left col-sm-6 col-xs-12 paddingZero">
                    <div class="contianer_icon">
                        <i class="fa fa-map-marker" aria-hidden="true"></i>
                    </div>
                    <div class="select_box">
                        <angular2-multiselect [data]="cityDropdownList" [(ngModel)]="selectedLocation" [settings]="cityDropdownSettings" (onSelect)="onItem1Select($event)" (onDeSelect)="OnItem1DeSelect($event)"></angular2-multiselect>
                    </div>
                </div>'

**Comment by CuppaLabs** [comment]
It might be an issue with the webpack or Your data structure might not be in the following structure. 
[
                          {"id":1,"itemName":"India"},
                          {"id":2,"itemName":"Singapore"},
                          {"id":3,"itemName":"Australia"},
                          {"id":4,"itemName":"Canada"}];

**Comment by Mridul16** [comment]
the strcuture is in the exactly same format

**Comment by Mridul16** [comment]
And I am already using one and it is working fine

**Comment by Mridul16** [comment]
the moment i use different [(ngModel)], it starts throwing error

**Comment by CuppaLabs** [comment]
Then the issue might be something else. Because in the demo page we are showing 8 dropdowns in one page without any issue. Please refer to the code from repo which might help you in fixing your issues.

**Comment by Mridul16** [comment]
I have done exactly the same thing. Yet i am facing this issue. Please help

**Comment by Mridul16** [comment]
Updated the module and its working now. Thanks.

**Comment by chamanklalwani** [comment]
Hi Mridul16,


I faced the similar issue and I fixed it in this way.

I have 3 dropdowns in my page. dd1 allows single selection other dd2 & dd3 allows multiple selection. On selecting a value from dd1, I am dynamically populating dd2 & dd3. My code started to break because of line#  3 of below-mentioned method "onItemSelect(item:any)"

**Solution:**

1) Don't assign "item" to "this.selectedItems" in your "onItemSelect()" & "onItem1Select()" method

onItemSelect(item:any) {
        console.log(item);
        console.log(this.selectedItems);
        this.selectedItems = item; // error was occuring because of this line in my code.
// because "item" is a simple json object and "this.selectedItems" expects an array or iterable.
    }
